### DISCUSSION TITLE
Moonraker Secrets

### DISCUSSION THREAD

**Discussion by jat08** [Arksine/moonraker#610] [Q&A] [answered]
Hi Everyone,

I am new to klipper, Mainsail and all that entails. I am trying to configure moonraker to send Pushover notifications but I cannot seem to get the first step right, configuring the secrets file. I have tried several variations and they all yield the same result:

Moonraker warning
Failed to load notifier[notifier pushover_start]
'components.secrets.Secrets object' has no attribute 'pushover_creds'
![image](https://user-images.githubusercontent.com/25801103/218232048-3a7417d0-2d1e-499e-a57d-80b067f980de.png)

Currently, my configuration looks as follows:

#######################################
### moonraker_secrets.ini
[pushover_creds]
token: 'my token'
user: 'my user'
device: 'my device'

#######################################
### pushover.conf
[notifier pushover_start]
url: pover://{secrets.pushover_creds.user}@{secrets.pushover_creds.token}/{secrets.pushover_creds.device}
events: started
title: Starting to print
body: Printing: {event_args[1].filename

#######################################
### moonraker.conf
[secrets]
[include pushover.conf]
#######################################

The moonraker_secrets.ini file is a file I manually created using WinSCP under that name and under the name of 'secret.ini' in a different attempt. I have tried creating that file under 'home/pi' directory, 'home/pi/printer_data' directory and under the manually created 'home/pi/printer_data/moonraker.secrets' directory, all with no luck.

I have tried just using the [secrets] section with no arguments under (as currently shown above), with no luck. But I have also tried with the 'path', data_path' and 'secrets_path' under the [secrets] section with no luck either. In fact, using any of these 'path' yield a different error as shown below:

Moonraker warning
Unparsed config option 'data_path: ~/secrets.ini' detected in section [secrets].
This may be an option no longer available or could be the result of a module that failed to load. In the future this will result in a startup error.
![image](https://user-images.githubusercontent.com/25801103/218231880-378972ac-6dac-4b5c-bcd4-b8987fb98215.png)

Can anyone help me with this? I would greatly appreciate it.

**Comment by Arksine** [comment, marked as answer]
You will need to download and attach `moonraker.log` for me to give a complete answer.  The secrets file should be named `moonraker.secrets` and placed in the data folder, which on a Raspberry Pi defaults to `/home/pi/printer_data`.  If you had to manually create the `printer_data` folder then you are likely using an outdated version of Moonraker.

**Reply by DrFate09** [reply]
@Arksine I am a bit confused about the correct procedure to setup a secrets file. In the Moonraker docs, it refers to the secrets file as "secrets.ini" (see image below) but here you refer to it as "moonraker.secrets".

So is the "moonraker.secrets" the actual file, or is it the folder that contains the "secrets.ini"?

<img width="785" height="640" alt="image" src="https://github.com/user-attachments/assets/33992a55-9e8e-43c2-874e-efaef2262aee" />

**Reply by Arksine** [reply]
The file name is `moonraker.secrets` and it should be directly in the data folder, ie: `~/printer_data/`.  It looks like the `secrets.ini` header on the example was mis-typed, I'll correct the docs.

**Comment by jat08** [comment]
Hi Arksine, thank you so much for your response. I renamed the file to 'moonraker.secrets' as you mentioned and that fixed the issue. Thanks for taking the time to help me, I really appreciate it.
